adapters/src/main/kotlin/com/krzykrucz/elesson/currentlesson/attendance/adapters/rest/AttendanceRouterConfig.kt

```diff
@@ -1,28 +1,76 @@
 package com.krzykrucz.elesson.currentlesson.attendance.adapters.rest
 
-import com.krzykrucz.elesson.currentlesson.attendance.domain.FetchCheckedAttendance
-import com.krzykrucz.elesson.currentlesson.attendance.domain.FetchNotCompletedAttendanceAndRegistry
-import com.krzykrucz.elesson.currentlesson.attendance.domain.PersistAttendance
+import arrow.core.Either
+import arrow.core.getOrElse
+import arrow.fx.IO
+import arrow.fx.typeclasses.Duration
+import com.krzykrucz.elesson.currentlesson.attendance.adapters.AttendanceDto
+import com.krzykrucz.elesson.currentlesson.attendance.adapters.AttendanceResponseDto
+import com.krzykrucz.elesson.currentlesson.attendance.adapters.LateAttendanceDto
+import com.krzykrucz.elesson.currentlesson.attendance.adapters.usecase.HandleNoteAbsent
+import com.krzykrucz.elesson.currentlesson.attendance.adapters.usecase.HandleNoteLate
+import com.krzykrucz.elesson.currentlesson.attendance.adapters.usecase.HandleNotePresent
+import com.krzykrucz.elesson.currentlesson.attendance.domain.AttendanceError
+import org.springframework.beans.factory.annotation.Qualifier
 import org.springframework.context.annotation.Bean
 import org.springframework.context.annotation.Configuration
 import org.springframework.http.MediaType
+import org.springframework.web.reactive.function.BodyInserters
+import org.springframework.web.reactive.function.server.ServerResponse
+import org.springframework.web.reactive.function.server.bodyToMono
 import org.springframework.web.reactive.function.server.router
+import reactor.core.publisher.Mono
+import java.util.concurrent.TimeUnit
 
 // TODO divide this config to 3 different adapters
 @Configuration
 class AttendanceRouterConfig {
 
     @Bean
-    fun attendanceRouter(
-            persistAttendance: PersistAttendance,
-            fetchNotCompletedAttendance: FetchNotCompletedAttendanceAndRegistry,
-            fetchCheckedAttendance: FetchCheckedAttendance
+    fun attendanceRouter(@Qualifier("noteAbsent") handleNoteAbsentDto: HandleNoteAbsent,
+                         @Qualifier("notePresent") handleNotePresentDto: HandleNotePresent,
+                         handleNoteLateDto: HandleNoteLate
     ) = router {
         (path("/attendance") and accept(MediaType.APPLICATION_JSON)).nest {
-            POST("/absent", handleNoteAbsentRequest(persistAttendance, fetchNotCompletedAttendance))
-            POST("/present", handleNotePresentRequest(persistAttendance, fetchNotCompletedAttendance))
-            POST("/late", handleNoteLateRequest(persistAttendance, fetchCheckedAttendance))
-            GET("", handleGetAttendanceRequest())
+            POST("/absent") { request ->
+                request.bodyToMono<AttendanceDto>()
+                    .flatMap { dto ->
+                        handleNoteAbsentDto(dto)
+                            .map { it.toServerResponse() }
+                            .run()
+                    }
+            }
+            POST("/present") { request ->
```

Comment:
it's interesting to approach to differentiate the purpose of the REST having `/attendance/absent`, it'd paid my attention earlier but I didn't mention it. What is the reason for such design? I more familiar with just single path `/attendance` and body which will give the purpose

Reply:
I thought that if the body will give the purpose for `attendance` resource there has to be some kind of forced difference between body for present and absent student. The idea behind current solution is that `attendance` has a resource `present` and I can post a student there or in the `absent` resource. I just wanted to avoid additional logic.